src/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java

```diff
@@ -133,7 +134,13 @@ public void scan(HttpMessage msg, String param, String value) {
 			boolean attackWorked = false;
             HttpMessage msg2 = getNewMsg();
 			setParameter(msg2, param, Constant.getEyeCatcher());
-            sendAndReceive(msg2);
+            try {
+    			sendAndReceive(msg2);
+        	} catch (InvalidRedirectLocationException|UnknownHostException e) {
+        		// Not an error, just means we probably attacked the redirect location

```

Comment:
So there are two attempts at the eyecatcher.
[L138](https://github.com/kingthorin/zap-extensions/blob/logging-and-exception-fixes/src/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java#L138), which I think we should allow to pass silently. If the second one fails [L164](https://github.com/kingthorin/zap-extensions/blob/logging-and-exception-fixes/src/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java#L164), then we can return out I guess. Seem reasonable?